### PR TITLE
Case insensitive operators

### PR DESCRIPTION
**Pull request by DeathByTape**
Added case insensitive operators: `prefixi`, `postfixi`, and `infixi`.

### CODE REVIEW

File: elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/FilterIT.groovy [hunk 1]
```diff
@@ -1015,6 +1017,96 @@ class FilterIT extends AbstractIntegrationTestInitializer {
...
+                RestAssured.get("/author/${nullNedId}/books?filter[book.editorName][postfixi]=D").asString())
+
+        Assert.assertEquals(editorEdBooks, editorNameEndsWithd.get("data").size())
+
+        /* Test RSQL Typed */
+        // TODO:
+//        editorNameEndsWithd = mapper.readTree(
+//                RestAssured.get("/author/${nullNedId}/books?filter[book]=genre==*Fiction").asString())
+//
+//        Assert.assertEquals(editorEdBooks, editorNameEndsWithd.get("data").size())
```
Comment (archolewa) [inline elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/FilterIT.groovy]: Any particular reason we're commenting out this stuff?

Comment (DeathByTape) [inline elide-integration-tests/src/test/groovy/com/yahoo/elide/tests/FilterIT.groovy]: It has a temporary TODO. I'm awaiting a response from this ticket to know how to support this feature with RSQL:

https://github.com/jirutka/rsql-parser/issues/25

**Comment by archolewa** [comment]
👍

**Comment by micheljung** [comment]
I almost opened an issue for this today - what a great timing :)

**Comment by DeathByTape** [comment]
@micheljung that's correct. I've opened a ticket with RSQL here: https://github.com/jirutka/rsql-parser/issues/25.

Will wait a short time for their response. If we don't hear back soon we will probably make a decision on how best to support this for now and contribute it back to the RSQL spec.

**Pull request merged**